src/openrct2/ride/Ride.cpp

```diff
@@ -6977,8 +6977,9 @@ void sub_6CB945(Ride* ride)
             const rct_preview_track* trackBlock = get_track_def_from_ride(ride, tileElement->AsTrack()->GetTrackType());
             while ((++trackBlock)->index != 0xFF)
             {
-                LocationXYZ16 blockLocation = location;
-                map_offset_with_rotation(&blockLocation.x, &blockLocation.y, trackBlock->x, trackBlock->y, direction);
+                auto blockLocationXY = map_offset_with_rotation(
```

Comment:
As per the other functions just do the sum and rotate directly. You should be able to update the type of location whilst you are at it to CoordsXYZ.